super_dpf.py

```diff
@@ -1,127 +1,316 @@
+            logging.info('Initializing config...')
+            with open(self.PATHS['conf_template'], 'r') as template:
+                template = json.load(template)
+                with open(self.PATHS['config'], 'w') as config_file:
+                    json.dump(template, config_file)
+
+            self._create_photo_dirs()
+
+    def _create_photo_dirs(self):
+        if not os.path.isdir(self.PATHS['photos']):
+            os.makedirs(self.PATHS['photos'])
+
+    def _walk_json_tree(self, node, kwargs):
+        for key, val in node.items():
+            if isinstance(val, dict):
+                self._walk_json_tree(val, kwargs)
+            elif key in kwargs:
+                node[key] = kwargs.get(key)
+            else:
+                continue
+        return node
+
+
+class LogManager(object):
+    LOG_LOCATION = os.path.dirname(os.path.realpath(__file__))
+    LOGGING = {
+        'version': 1,
+        'formatters': {
+            'standard': {
+                'format': "[%(asctime)s] %(levelname)s [%(name)s:%(lineno)s] %(message)s",
+                'datefmt': "%d/%b/%Y %H:%M:%S"
+            },
+            'verbose': {
+                'format': 'dpf[%(process)d]: %(levelname)s %(name)s[%(module)s] %(message)s'
+            },
+        },
+        'handlers': {
+            'null': {
+                'level': 'DEBUG',
+                'class': 'logging.NullHandler',
+            },
+            # 'mail_admins': {
+            #     'level': 'ERROR',
+            #     'class': 'logging.handlers.AdminEmailHandler'
+            # },
+            'logfile': {
+                'level': 'INFO',
+                'class': 'logging.handlers.RotatingFileHandler',
+                'filename': '{}/sdpf.log'.format(LOG_LOCATION),
+                'maxBytes': 50000000,
+                'backupCount': 2,
+                'formatter': 'standard',
+            },
+            'console': {
+                'level': 'DEBUG',
+                'class': 'logging.StreamHandler',
+                'formatter': 'standard'
+            },
+            'syslog': {
+                'level': 'INFO',
+                'class': 'logging.handlers.SysLogHandler',
+                'formatter': 'verbose',
+                'address': '/dev/log',
+                'facility': 'local2',
+            }
+        },
+        'loggers': {
+            'django': {
+                'handlers': ['console', 'syslog'],
+                'propagate': True,
+                'level': 'WARN',
+            },
+            'urllib3': {
+                'handlers': ['console', 'logfile', 'syslog'],
+                'propagate': True,
+                'level': 'ERROR',
+            },
+            '': {
+                'handlers': ['console', 'logfile', 'syslog'],
+                'propagate': True,
+                'level': 'DEBUG',
+            },
+        }
+    }
 
+    dictConfig(LOGGING)
+
+
+class BaseDPF(DPFConfigurator, LogManager):
```

Comment:
When I suggested that DPFConfigurator be a distinct class I did not intend suggesting a mixin that is eventually inherited all the way down to a specific DPF subclass.  The configuration needs to support multiple subclasses so it logically needs to manage configuration in a global scope while being able to invoke configuration dialog methods of DPF subclasses.

It might make sense for the config dialogs to be classmethods on each DPF subclass.  When instantiating a DPF subclass the configuration that was handed back from the config class could then be passed to the constructor.  This way the class instance only has the configuration details it cares about and knew how to create instead of having to sift through the whole config.